website-client/src/components/Controller/parse-command.ts

```diff
@@ -18,6 +18,99 @@ const buttonNameToStateMember: { [buttonName: string]: string } = {
 	right: 'arrowRight',
 }
 
+/**
+ * Updates `controllerState`.
+ * @param command A single command that can be represented by one state such as just pressing a button down or just letting a button back up but not tapping a button (e.g. 'x' which involves pushing down then back up). Can use '&' to combine commands.
+ * @param controllerState The current state.
+ * @param fullCommand The full command that `singleCommand` came from.
+ */
+function updateState(command: string, controllerState: ControllerState, fullCommand?: string): void {
+	if (fullCommand === undefined) {
+		fullCommand = command
+	}
+	for (let singleCommand of command.split('&')) {
+		singleCommand = singleCommand.trim()
+		const commandParts = singleCommand.split(/\s+/)
+		if (commandParts.length < 2) {
+			// A button might be tapped. Not really supported but it should not update the state.
+			console.warn("Ignoring unrecognized part of command: \"%s\" from \"%s\"", singleCommand, fullCommand)
```

Comment:
To make it clear that the command will still get sent, these warnings should be clarified to say that the state will not be updated for that part.